db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/InListOperatorNode.java

```diff
@@ -917,52 +917,81 @@ public void generateStartStopKey(boolean isAsc, boolean isStartKey,
-	 * in DESCENDING order.  This only applies to in-list "multi-probing",
-	 * where the rows are processed in the order of the IN list elements
-	 * themselves.  In that case, any requirement to sort the rows in
-	 * descending order means that the values in the IN list have to
-	 * be sorted in descending order, as well.
-	 */
-	protected void markSortDescending()
-	{
-		sortDescending = true;
-	}
-
-	/**
-	 * Return whether or not the IN-list values for this node are ordered.
-	 * This is used for determining whether or not we need to do an execution-
-	 * time sort.
-	 */
-	protected boolean isOrdered()
-	{
-		return isOrdered;
-	} 
-
-	/**
-	 * Return whether or not the IN-list values for this node must be
-	 * sorted in DESCENDING order.
-	 */
-	protected boolean sortDescending()
-	{
-		return sortDescending;
-	}
-	
-	@Override
-	public int hashCode(){
-		int result = super.hashCode();
-		return Objects.hash(result, isOrdered, sortDescending);
-	}
+    }
+
+    /**
+     * Indicate that the IN-list values for this node are ordered (i.e. they
+     * are all constants and they have been sorted).
+     */
+    protected void markAsOrdered(boolean ordered)
+    {
+        isOrdered = ordered;
+    }
+
+    /**
+     * Indicate that the IN-list values for this node must be sorted
+     * in DESCENDING order.  This only applies to in-list "multi-probing",
+     * where the rows are processed in the order of the IN list elements
+     * themselves.  In that case, any requirement to sort the rows in
+     * descending order means that the values in the IN list have to
+     * be sorted in descending order, as well.
+     */
+    protected void markSortDescending()
+    {
+        sortDescending = true;
+    }
+
+    /**
+     * Return whether or not the IN-list values for this node are ordered.
+     * This is used for determining whether or not we need to do an execution-
+     * time sort.
+     */
+    protected boolean isOrdered()
+    {
+        return isOrdered;
+    }
+
+    /**
+     * Return whether or not the IN-list values for this node must be
+     * sorted in DESCENDING order.
+     */
+    protected boolean sortDescending()
+    {
+        return sortDescending;
+    }
+
+    @Override
+    public int hashCode(){
+        int result = super.hashCode();
+        return Objects.hash(result, isOrdered, sortDescending);
+    }
+
+    ValueNode evaluateConstantExpressions() throws StandardException {
+        boolean DB2CompatibilityMode = getCompilerContext().getVarcharDB2CompatibilityMode();
+        if (DB2CompatibilityMode) {
+            return this;
+        }
+        if (!(getLeftOperand() instanceof ConstantNode)) {
+            return this;
+        }
+        if (!getRightOperandList().containsAllConstantNodes()) {
+            return this;
+        }
+
+        DataValueDescriptor lhs = ((ConstantNode) getLeftOperand()).getValue();
+        if (lhs == null || lhs.isNull()) {
+            return this;
```

Comment:
I think we can return `false` here, right?
i.e.
`return (ValueNode) getNodeFactory().getNode(C_NodeTypes.BOOLEAN_CONSTANT_NODE, false, getContextManager());`